### PR TITLE
Remove unnecessary Evergreen deps

### PR DESCRIPTION
**Pull request by kaidokert**
Further reduce unnecessary Evergreen dependency coupling.

b/331996918

### CODE REVIEW

File: starboard/raspi/shared/BUILD.gn [hunk 1]
```diff
@@ -378,8 +378,13 @@ static_library("starboard_base_symbolize") {
     "//third_party/symbolize/demangle.cc",
     "//third_party/symbolize/symbolize.cc",
   ]
-
-  public_deps = [ "//starboard/elf_loader:evergreen_info" ]
+  deps = [
```
Comment (niranjanyardi) [inline starboard/raspi/shared/BUILD.gn:381]: if things were building without this , why was this added now ?

Comment (kaidokert) [reply starboard/raspi/shared/BUILD.gn:381]: Because the symbolize.cc uses  a few starboard headers, and previously those were pulled in transitively via elf_loader.
